docs/source/python/filesystems.rst

```diff
@@ -200,10 +240,34 @@ For example::
    import pyarrow.dataset as ds
    ds.dataset("data/", filesystem=fs)
 
+::
+
+   # Similarly for Azure Blob Storage
+   import adlfs
+   # ... load your credentials and configure the filesystem
+   fs = adlfs.AzureBlobFileSystem(account_name=account_name, account_key=account_key)
+
+   import pyarrow.dataset as ds
+   ds.dataset("data/", filesystem=fs)
```

Comment:
Nice! Did you check this actually works?

Reply:
Yes I have just tried with a couple of partitions in an azure container (the equivalent of s3 bucket).
I have added the "container" bit in a recent commit to make that more clear.